### PR TITLE
[Data] Update Data quickstart on Ray "Getting Started"

### PR DESCRIPTION
**Pull request by bveeramani**
## Why are these changes needed?

The existing quickstart is verbose. This PR simplifies it.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-overview/getting-started.md [hunk 1]
```diff
@@ -94,49 +94,48 @@ Learn more about Ray AIR
...
-Use individual libraries for single ML workloads, without having to install the full AI Runtime package. Click on the dropdowns for your workload below. 
+Use individual libraries for single ML workloads, without having to install the full AI Runtime package. Click on the dropdowns for your workload below.
 
 `````{dropdown} <img src="images/ray_svg_logo.svg" alt="ray" width="50px"> Data: Scalable Datasets for ML
 :animate: fade-in-slide-down
 
-Ray Data is the standard way to load and exchange data in Ray libraries and applications.
-Ray Data provides basic distributed data transformations such as `map`, `filter`, and `repartition`.
-They are compatible with a variety of file formats, datasources, and distributed frameworks.
+Scale offline inference and model training with [Ray Data](data_key_concepts) --
```
Comment (amogkam) [inline doc/source/ray-overview/getting-started.md:102]: ingest for training, not model training itself

File: doc/source/ray-overview/getting-started.md [hunk 2]
```diff
@@ -94,49 +94,48 @@ Learn more about Ray AIR
...
-:start-after: __create_from_python_begin__
-:end-before: __create_from_python_end__
-```
+# Create datasets from on-disk files, Python objects, and cloud storage like S3.
+ds = ray.data.read_csv("s3://anonymous@ray-example-data/iris.csv")
 
-Datasets can be created from files on local disk or remote datasources such as S3. Any filesystem 
-[supported by pyarrow](http://arrow.apache.org/docs/python/generated/pyarrow.fs.FileSystem.html) can be used to specify file locations.
-You can also create a ``Dataset`` from existing data in the Ray object store or Ray-compatible distributed DataFrames:
+# Apply functions to transform data. Ray executes transformations in parallel.
```
Comment (amogkam) [inline doc/source/ray-overview/getting-started.md:124]:
```suggestion
# Apply functions to transform data. Ray Data executes transformations in parallel.
```

File: doc/source/ray-overview/getting-started.md [hunk 3]
```diff
@@ -94,49 +94,48 @@ Learn more about Ray AIR
...
-Datasets also supports ``.filter()`` and ``.flat_map()``.
+transformed_ds = ds.map_batches(compute_area)
 
-```{literalinclude} ../data/doc_code/quick_start.py
-:language: python
-:start-after: __data_transform_begin__
-:end-before: __data_transform_end__
+# Iterate over batches of data.
+for batch in transformed_ds.iter_batches(batch_size=4):
+    pass
```
Comment (amogkam) [inline doc/source/ray-overview/getting-started.md:135]: do something other than pass?

File: doc/source/ray-overview/getting-started.md [hunk 4]
```diff
@@ -94,49 +94,48 @@ Learn more about Ray AIR
...
-```{literalinclude} ../data/doc_code/quick_start.py
-:language: python
-:start-after: __data_transform_begin__
-:end-before: __data_transform_end__
+# Iterate over batches of data.
+for batch in transformed_ds.iter_batches(batch_size=4):
+    print(batch)
+
+# Save dataset contents to on-disk files or cloud storage.
+transformed_ds.write_parquet("/tmp/iris/")
```
Comment (amogkam) [inline doc/source/ray-overview/getting-started.md:138]: use local scheme?

**Review by amogkam — APPROVED** [review]

**Review by ericl — APPROVED** [review]

**Pull request merged**